amr-wind/core/FieldBCOps.H

```diff
@@ -203,11 +221,17 @@ struct DirichletOp
 
 struct FieldBCDirichlet
 {
-    using FunctorType = DirichletOp<ConstDirichlet, ConstDirichlet>;
+    using FunctorType = DirichletOp<ConstDirichlet, ConstDirichlet, &CC_idx>;
+    using FunctorXType = DirichletOp<ConstDirichlet, ConstDirichlet, &FX_idx>;
+    using FunctorYType = DirichletOp<ConstDirichlet, ConstDirichlet, &FY_idx>;
+    using FunctorZType = DirichletOp<ConstDirichlet, ConstDirichlet, &FZ_idx>;
 
     explicit FieldBCDirichlet(const Field& fld) : m_field(fld) {}
 
     inline FunctorType operator()() const { return FunctorType(m_field); }
+    inline FunctorXType opfx() const { return FunctorXType(m_field); }
```

Comment:
I gotta think there's a way to template these... `template <typename T> T opf() const {return T(m_field);}`. Can you see if that's possible?

Reply:
I didn't include FunctorType in the templating but that is definitely possible too.